server/src/main/scala/controllers/application/WithStaffing.scala

```diff
@@ -80,4 +87,102 @@ trait WithStaffing {
     }
   }
 
+  def addStaffMovements(): Action[AnyContent] = authByRole(StaffMovementsEdit) {
+    Action {
+      request =>
+        request.body.asText match {
+          case Some(text) =>
+            val movementsToAdd: List[StaffMovement] = read[List[StaffMovement]](text)
+            ctrl.staffMovementsActor ! AddStaffMovements(movementsToAdd)
+            Accepted
+          case None =>
+            BadRequest
+        }
+    }
+  }
+
+
+  def removeStaffMovements(movementsToRemove: UUID): Action[AnyContent] = authByRole(StaffMovementsEdit) {
+    Action {
+
+      ctrl.staffMovementsActor ! RemoveStaffMovements(movementsToRemove)
+      Accepted
+    }
+  }
+
+  def getStaffMovements(maybePointInTime: Option[MillisSinceEpoch]): Action[AnyContent] = authByRole(BorderForceStaff) {
+    Action.async {
+      val staffMovementsFuture = maybePointInTime match {
+        case None =>
+          ctrl.staffMovementsActor.ask(GetState)
+            .map { case StaffMovements(movements) => movements }
+            .recoverWith { case _ => Future(Seq()) }
+
+        case Some(millis) =>
+          val date = SDate(millis)
+
+          val actorName = "staff-movements-read-actor-" + UUID.randomUUID().toString
+          val staffMovementsReadActor: ActorRef = system.actorOf(Props(classOf[StaffMovementsReadActor], date, DrtStaticParameters.time48HoursAgo(() => date)), actorName)
+
+          staffMovementsReadActor.ask(GetState)
+            .map { case StaffMovements(movements) =>
+              staffMovementsReadActor ! PoisonPill
+              movements
+            }
+            .recoverWith {
+              case _ =>
+                staffMovementsReadActor ! PoisonPill
+                Future(Seq())
+            }
+      }
+
+      val eventualStaffMovements = staffMovementsFuture.collect {
+        case Nil =>
+          log.debug(s"Got no movements")
+          List()
+        case sm: Seq[StaffMovement] => sm
+      }
+
+      eventualStaffMovements.map(sms => Ok(write(sms)))
+    }
+  }
+
+  def exportStaffMovements(terminalString: String, pointInTime: MillisSinceEpoch): Action[AnyContent] =
+    authByRole(StaffMovementsExportRole) {
+      Action {
+        val terminal = Terminal(terminalString)
+        val date = SDate(pointInTime)
+        val actorName = "staff-movements-read-actor-" + UUID.randomUUID().toString
+        val staffMovementsReadActor: ActorRef = system.actorOf(Props(classOf[StaffMovementsReadActor], date, DrtStaticParameters.time48HoursAgo(() => date)), actorName)
+
+        val eventualStaffMovements: Future[Seq[StaffMovement]] = staffMovementsReadActor.ask(GetState)
```

Comment:
shall we extract this ask, map & recover this to a function as it's repeated in the previous function?